### ISSUE TITLE
presign url doesn't seem to generate a token

### ISSUE THREAD

**Issue by silky** [brendanhay/amazonka#840]
no matter what i do, using the latest `main` branch, i can't convince `presignURL` to generate a token; and as such it fails to authenticate

here's what i get, roughly:

```
http://localhost:9000/abstraction.png
  ?X-Amz-Algorithm=AWS4-HMAC-SHA256
  &X-Amz-Credential=key%2F20221123%2Fus-east-1%2Fs3%2Faws4_request
  &X-Amz-Date=20221123T174416Z
  &X-Amz-Expires=900
  &X-Amz-SignedHeaders=host
  &X-Amz-Signature=...
```

note a distinct lack of `X-Amz-Token`

apparently it used to do this, according to google. but not anymore?

**Comment by endgame** [comment]
Can you please post the code you're using to generate this presigned URL?

**Comment by endgame** [comment]
Also, do you mean `X-Amz-Security-Token` and are you using temporary credentials to sign this URL? There is no `X-Amz-Token` on https://docs.aws.amazon.com/AmazonS3/latest/API/sigv4-query-string-auth.html

**Comment by silky** [comment]
thanks for the quick response!

here's the exact code:

```
bucket' :: BucketName
bucket' = BucketName "hello"


object' :: ObjectKey
object' = ObjectKey "abstraction.png"


main :: IO ()
main = do
  void $ loadFile defaultConfig

  let ourS3 = setEndpoint False "localhost" 9000 defaultService

  env  <- newEnv Discover <&> configure ourS3
  time <- getCurrentTime

  let p = newGetObject bucket' object'

  url <- runResourceT $ do
      presignURL env time 900 p

  putStrLn $ C.unpack url
```

in my env I have 

```
AWS_ACCESS_KEY_ID=key
AWS_SECRET_ACCESS_KEY=...
```

these are credentials that i made using minio's "access key" interface.

the reason i think it should have a token is i get this error upon using the resulting one:

![image](https://user-images.githubusercontent.com/129525/203724137-7a4f7171-6b95-4f00-a5d3-81515e38187b.png)

( note i changed the url by hand to get the bucket right, as i haven't fixed it based on your feedback on my other issue - #841 - yet, thanks for that! :) )

...

and for example here's the url minio itself generates:

```
http://127.0.0.1:9000/hello/abstraction.png
  ?X-Amz-Algorithm=AWS4-HMAC-SHA256
  &X-Amz-Credential=Q58EK2R7A16DRS4FFG3Q%2F20221124%2Fus-east-1%2Fs3%2Faws4_request
  &X-Amz-Date=20221124T074811Z
  &X-Amz-Expires=604800
  &X-Amz-Security-Token=eyJhbGciOiJIU....
  &X-Amz-SignedHeaders=host
  &versionId=null
  &X-Amz-Signature=05f1caa3a...
```

maybe it's just a problem with how i've configured minio? but as far as i can tell, it doesn't look like presign-url even hits the service; i.e. if i get the password/key wrong, it still generates a url.

**Comment by silky** [comment]
interesting; i think it works now that i incorporated your fix for #841 ;

thanks for the help!
